Composer/packages/electron-server/src/main.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { join, resolve } from 'path';
+
+import { mkdirp } from 'fs-extra';
+import { app, BrowserWindow } from 'electron';
+import fixPath from 'fix-path';
+
+import { getUnpackedAsarPath } from './utility/getUnpackedAsarPath';
+
+const isDevelopment = process.env.NODE_ENV === 'development';
+
+function main() {
+  console.log('Starting electron app');
+  // Create the browser window.
+  const browserWindowOptions: Electron.BrowserWindowConstructorOptions = {
+    width: 800,
+    height: 600,
+    webPreferences: {
+      nodeIntegration: true,
+    },
+    show: false,
+  };
+  if (process.platform === 'linux' && !isDevelopment) {
+    // workaround for broken .AppImage icons since electron-builder@21.0.1 removed .AppImage desktop integration
+    // (https://github.com/electron-userland/electron-builder/releases/tag/v21.0.1)
+    browserWindowOptions.icon = join(getUnpackedAsarPath(), 'resources/composerIcon_1024x1024.png');
+  }
+  const win = new BrowserWindow(browserWindowOptions);
+
+  // and load the index.html of the app.
+  const CONTENT_URL = isDevelopment ? 'http://localhost:3000/' : 'http://localhost:5000/';
+  console.log('Loading project from: ', CONTENT_URL);
+
+  // TODO: get rid of this timeout
+  setTimeout(() => {
```

Comment:
when and why can this be removed?

Reply:
This apparently was an issue on Mac. @srinaath can provide more context. This was not required on Windows.

Reply:
So this needs to be removed hopefully soon. So, the reason is we `await` for the `server` to spin up. Electron builds a shell around `http://localhost:5000`. Server start() resolves the promise yet it takes a couple of seconds before `http://localhost:5000` is available. Hence, if i dont put that timeout, electron waits for the promise, the promise is resolved and immediately it spins up the wrapper window around the URL which is still empty as `http://localhost:5000` is not available yet (takes a few seconds for it to load on Mac). So, we have to  refresh contents after the window is loaded for us to get the correct view. I have tried a few electron events that indicate if the url is loaded or not etc. None seem to work. But, I figured this is unblocking our distribution pipeline for now  and I can work on removing it in the next few PR's

Reply:
Ah. I think I see why. In `server.ts` just return a promise and resolve when the express app is ready. Line 129.

Reply:
Thanks Andy.. Tony has made the fix to remove it